cscs-checks/libraries/gridtools/gridtools_base.py

```diff
@@ -0,0 +1,180 @@
+#                                         '-DCMAKE_EXPORT_COMPILE_COMMANDS=ON',
+#                                         '-DBOOST_ROOT=$BOOST_ROOT',
+#                                         '-DGT_ENABLE_PYUTILS=OFF',
+#                                         '-DGT_TESTS_REQUIRE_FORTRAN_COMPILER=ON',
+#                                         '-DGT_TESTS_REQUIRE_C_COMPILER=ON',
+#                                         '-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON']
+        self.build_system.config_opts = ['-DCMAKE_INSTALL_PREFIX=/scratch/snx3000/bignamic/reframe/cscs-checks/libraries/gridtools/install',
+                                         '-DBoost_NO_BOOST_CMAKE="true"',
+                                         '-DCMAKE_BUILD_TYPE:STRING=Release',
+                                         '-DBUILD_SHARED_LIBS:BOOL=ON',
+                                         '-DGT_ENABLE_TARGET_X86:BOOL=OFF',
+                                         '-DGT_ENABLE_TARGET_NAIVE:BOOL=OFF',
+                                         '-DGT_ENABLE_TARGET_CUDA:BOOL=ON',
+                                         '-DCUDA_ARCH:STRING=sm_60',
+                                         '-DGT_GCL_ONLY:BOOL=OFF',
+                                         '-DCMAKE_CXX_COMPILER=CC',
+                                         '-DCMAKE_CUDA_HOST_COMPILER:STRING=CC',
+                                         '-DGT_USE_MPI:BOOL=OFF', 
+                                         '-DGT_SINGLE_PRECISION:BOOL=OFF', 
+                                         '-DGT_ENABLE_PERFORMANCE_METERS:BOOL=ON',
+                                         '-DGT_TESTS_ICOSAHEDRAL_GRID:BOOL=OFF',
+                                         '-DCMAKE_EXPORT_COMPILE_COMMANDS=ON',
+                                         '-DBOOST_ROOT=$BOOST_ROOT',
+                                         '-DGT_ENABLE_PYUTILS=OFF',
+                                         '-DGT_TESTS_REQUIRE_FORTRAN_COMPILER=ON',
+                                         '-DGT_TESTS_REQUIRE_C_COMPILER=ON',
+                                         '-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON']
+
+
+        self.sanity_patterns = sn.assert_found(r'PASSED', self.stdout)
+        self.perf_patterns = {
+            'perf': sn.extractsingle(r'(?P<timer>\w+) ms total',
+                                     self.stdout, 'timer', int)
+        }
+        self.build_system.max_concurrency = 2
+        self.tags = {'production'}
+        self.maintainers = ['CB']
+
+@rfm.parameterized_test(['VerticalAdvectionNaive'], ['VerticalAdvectionMC'], 
+                        ['SimpleHoriDiffNaive'], ['SimpleHoriDiffMC'])
+class GridToolsCheckCPU(GridToolsCheckBase):
+    def __init__(self, test_name):
+        super().__init__()
+        self.descr = 'GridTools host test'
+
+        self.valid_systems = ['daint:mc', 'daint:gpu']
+        self.num_gpus_per_node = 0
+        self.num_tasks = 1
+
+        if test_name == 'VerticalAdvectionNaive':
```

Comment:
Why don't you use as test names the underscore variables so that you can easily construct your variables later on?